osquery/tables/system/windows/handles.cpp

```diff
@@ -0,0 +1,280 @@
+    }
+    free(objectTypeInfo);
+
+    return Status::failure("Could not get object name for " + std::get<0>(objInfo));
+}
+
+Status getSystemHandles(PSYSTEM_HANDLE_INFORMATION &handleInfo) {
+    NTSTATUS ntstatus;
+    ULONG handleInfoSize = 0x1000;
+
+    auto _NtQuerySystemInformation = reinterpret_cast<NtQuerySystemInformation>(GetProcAddress(GetModuleHandleA("ntdll.dll"), "NtQuerySystemInformation"));
+
+    if ((handleInfo = (PSYSTEM_HANDLE_INFORMATION)malloc(handleInfoSize)) == NULL){
+        return Status(GetLastError(), "Could not allocate memory for handleInfo");
+    }
+
+	while ((ULONG)(ntstatus = _NtQuerySystemInformation(SystemHandleInformation, handleInfo, handleInfoSize, NULL)) == STATUS_INFO_LENGTH_MISMATCH)
+    {
+        if ((handleInfo = (PSYSTEM_HANDLE_INFORMATION)realloc(handleInfo, handleInfoSize *= 2)) == NULL){
+            return Status(GetLastError(), "Could not re-allocate memory for for handleInfo");
+        }
+    }
+    if (ntstatus != STATUS_SUCCESS ) {
+        return Status(ntstatus, "Could not get system informations");
+    }
+    return Status::success();
+}
+
+QueryData genHandles(QueryContext &context) {
+    QueryData rows;
+    ULONG   i;
+    DWORD   currentPid;
+    HANDLE  processHandle;
+    HANDLE  processDupHandle;
+    PSYSTEM_HANDLE_INFORMATION handleInfo;
+    SYSTEM_HANDLE_TABLE_ENTRY_INFO  handle;
+
+    // HMODULE ntdllModule;
+    auto _NtDuplicateObject = reinterpret_cast<NtDuplicateObject>(GetProcAddress(GetModuleHandleA("ntdll.dll"), "NtDuplicateObject"));
+    auto _NtQueryObject = reinterpret_cast<NtQueryObject>(GetProcAddress(GetModuleHandleA("ntdll.dll"), "NtQueryObject"));
+
+    auto status = getSystemHandles(handleInfo);
+    if (!status.ok()) {
+        VLOG(1) << L"Unable to get system handles: " << status.getCode();
+        return rows;
+    }
+
+    currentPid = GetCurrentProcessId();
+    for (i = 0; i < handleInfo->NumberOfHandles; i++)
+    {
+        if (handleInfo->Handles[i].UniqueProcessId == currentPid) {
+            // Do not get handles for the current process
+            continue;
+        }
+        handle = handleInfo->Handles[i];
+	    if (!(processHandle = OpenProcess(PROCESS_DUP_HANDLE, FALSE, handle.UniqueProcessId))) {
+            continue;
+        }
+        if (_NtDuplicateObject(
+            processHandle,
+            (HANDLE)(intptr_t)handle.HandleValue,
+            GetCurrentProcess(),
+            &processDupHandle,
+            GENERIC_READ,
+            0,
+            0) != STATUS_SUCCESS)
+        {
+            CloseHandle(processHandle);
+            CloseHandle(processDupHandle);
+            continue;
+        }
+
+        ObjectInfoTuple objInfo;
+        auto status = getHandleInfo(processDupHandle, _NtQueryObject, objInfo);
+        if (!status.ok()) {
+            VLOG(1) << status.getMessage();
+            continue;
+        }
+        // Build a row from the provided handle informations
+        Row r;
```

Comment:
I've added the `attributes` to the virtual table, but i'm not sure if its the cleanest way to do it.
Also, I could not figure out how to properly parse the `ACCESS_MASK`. If you have any clue on this subject i would be happy, otherwise this can be left out as a feature request.